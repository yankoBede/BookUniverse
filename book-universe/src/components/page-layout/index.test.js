import React from 'react'
import renderer from 'react-test-renderer'
import TestingEnvironment from '../../test-utils/router'
import PageLayout from './index'
jest.mock('../header', () => 'Header')
jest.mock('../footer', () => 'Footer')

describe('Page lAYOUT Component', () => {
    it('should render authenticated routes', () => {
        const tree = renderer.create(
            <TestingEnvironment value= {{
                user: {
                    loggedIn: true,
                    id: '123'
                }
            }}>
                <PageLayout/>
            </TestingEnvironment>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })


    it('should render non-authenticated routes', () => {
        const tree = renderer.create(
            <TestingEnvironment value= {{
                user: {
                    loggedIn: false
                }
            }}>
                <PageLayout/>
            </TestingEnvironment>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})