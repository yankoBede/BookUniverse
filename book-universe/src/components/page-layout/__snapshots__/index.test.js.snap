// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page lAYOUT Component should render authenticated routes 1`] = `
<div
  className="app"
>
  <Header />
  <div
    className="container"
  >
    <div
      className="inner-container"
    />
  </div>
  <Footer />
</div>
`;

exports[`Page lAYOUT Component should render non-authenticated routes 1`] = `
<div
  className="app"
>
  <Header />
  <div
    className="container"
  >
    <div
      className="inner-container"
    />
  </div>
  <Footer />
</div>
`;
